src/analysis/Environments/Environments.models.ts

```diff
@@ -22,5 +22,9 @@ export interface LeoResourcePermissionsProvider {
   canPauseResource: (resource: App | ListRuntimeItem) => boolean;
 }
 
+export interface CanDeleteProvider {
+  canBeDeleted: (resource: App) => boolean;
```

Comment:
I'd also call it `canDeleteApp` to be consistent with other interface method names